macros/src/lib.rs

```diff
@@ -204,7 +197,8 @@ pub fn timestamp(ts: TokenStream) -> TokenStream {
                 // Unique symbol name to prevent multiple `timestamp!` invocations in the crate graph.
                 // Uses `#symname` to ensure it is not discarded by the linker.
                 #[no_mangle]
-                #[link_section = #section]
+                #[cfg_attr(target_os = "macos", link_section = ".defmt,end_timestamp")]
+                #[cfg_attr(not(target_os = "macos"), link_section = ".defmt.end_timestamp")]
```

Comment:
This needs to use `end.timestamp` because the linker script special-cases the `end` subsection

Reply:
for non-macos only though? Otherwise we need to change the special casing, macos won't allow any dots except the leading one

Reply:
I spoke too soon - `.defmt,end.timestamp` seems fine under macos, unlike `.defmt.end.timestamp`...?

Reply:
Ah, yeah, for non-macOS should be enough then